### PR TITLE
feat(TreeView): TreeView and TreeItem components

### PR DESCRIPTION
**Pull request by acoopa**
Issue: #180 

## What I did

Created a TreeView component.

## Screenshots:

image
image
image

## Checklist 
- [X] changeset has been added
- [X] Pull request description is descriptive
- [X] I have made corresponding changes to the documentation
- [X] New and existing unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## How to test

Requirements and testing notes found in Notion: https://laura-silva.notion.site/TreeView-1fa9fc9d40654f7dba969a7fbaa492c6

### CODE REVIEW

File: packages/react-magma-dom/src/components/TreeView/useTreeView.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import * as React from "react"
+
+export enum ExpandInitialOptions {
+  all = "all",
+  first = "first",
+  none = "none",
+}
+export interface UseTreeViewProps {
+  expandInitial?: ExpandInitialOptions;
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/TreeView/useTreeView.ts:9]: Can we add a description for these props, and outline the default value?

File: packages/react-magma-dom/src/components/TreeView/index.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// export * from "./useTreeView"
```
Comment (silvalaura) [inline packages/react-magma-dom/src/components/TreeView/index.ts:1]: Do we need to uncomment this line?

**Review by silvalaura — COMMENTED** [review]
Looking great so far, nice work! 👏 

A couple other notes:
- I noticed that the expand buttons don't look clickable (the cursor doesn't change to the pointer). Also, the accessibility check on storybook calls out `Buttons must have discernible text` so I think they are just missing the aria label attribute
- We try not to commit the package-lock unless something has changed in our packages

**Review by ccedrone — APPROVED** [review]

**Comment by silvalaura** [comment]
Although there are some outstanding TODOs, there will be a followup PR to clean that up.

**Pull request merged**